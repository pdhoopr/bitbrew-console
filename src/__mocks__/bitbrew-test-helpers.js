import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import diff from 'jest-diff';
import { addSerializer, getSerializers, utils } from 'jest-snapshot';
import styledSerializer from 'jest-styled-components/src/styleSheetSerializer';
import stripAnsi from 'strip-ansi';

const DIFF_HEADER = 'Diff:\n';

export const diffComponents = (baseComponent, compareComponent) => {
  const serializers = getSerializers();
  [styledSerializer, enzymeSerializer].forEach(serializer => {
    if (!serializers.includes(serializer)) {
      addSerializer(serializer);
    }
  });
  const base = `${utils.serialize(shallow(baseComponent))}\n`;
  const compare = `${utils.serialize(shallow(compareComponent))}\n`;
  const changes = diff(base, compare, {
    aAnnotation: `<BaseComponent />`,
    bAnnotation: `<CompareComponent />`,
    contextLines: 3,
    expand: false,
  });
  return `${DIFF_HEADER}${changes}`;
};

export const diffSerializer = {
  test(value) {
    return typeof value === 'string' && value.startsWith(DIFF_HEADER);
  },
  print(value) {
    return stripAnsi(value).trim();
  },
};

export const fixtures = {
  accessToken:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE1MjUxMTUzNTAsImV4cCI6MTU1NjY1MTM1MCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSJ9.4O-W8aeOOUFU8fkGTAFaAu9D9nK_l61yPomNG1Qkny8',
  orgs: {
    danlaw: {
      id: 'bd006616-3bd2-4783-a84d-b0fa4ed81135',
      properName: 'Danlaw',
    },
    azuga: {
      id: '8d9ec637-f3bb-4c46-83ee-11764d09f0b2',
      properName: 'Azuga',
    },
    bitbrew: {
      id: '94d9b0ca-a758-4502-8dcd-41d883e38714',
      properName: 'BitBrew',
    },
  },
  users: {
    bill: {
      email: 'bill@bitbrew.com',
      name: 'Bill Monaghan',
    },
    cynthia: {
      email: 'cynthia@bitbrew.com',
      name: 'Cynthia Hamilton',
    },
    alex: {
      email: 'bill@bitbrew.com',
      name: 'Alex Li',
    },
  },
};
