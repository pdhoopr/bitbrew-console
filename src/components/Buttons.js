import styled, { css } from 'styled-components';

export const baseStyles = css`
  background: none;
  border: none;
  border-radius: var(--corner-radius);
  cursor: pointer;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  font-weight: var(--weight-bold);
  line-height: var(--size-20);
  padding: var(--size-8) var(--size-16);
  text-transform: uppercase;
  transition: opacity var(--duration-short);

  &:hover {
    opacity: 0.8;
  }
`;

export const RaisedButton = styled.button`
  ${baseStyles};
  color: var(--color-white);
  background-color: var(--color-green);
  box-shadow: var(--elevation-low);
`;

export const TextButton = styled.button`
  ${baseStyles};
  color: inherit;
`;
