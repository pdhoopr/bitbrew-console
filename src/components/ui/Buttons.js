import styled, { css } from 'styled-components';

const baseStyles = css`
  appearance: button;
  background: none;
  border: none;
  border-radius: var(--corner-radius);
  cursor: pointer;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  font-weight: var(--weight-bold);
  line-height: var(--size-20);
  margin: 0;
  padding: var(--size-8) var(--size-16);
  text-transform: uppercase;

  &::-moz-focus-inner {
    border-style: none;
    padding: 0;
  }

  &:-moz-focusring {
    outline: 1px dotted ButtonText;
  }
`;

export const Button = styled.button.attrs({
  type: 'button',
})`
  ${baseStyles};
  color: inherit;
  padding: var(--size-8);
  position: relative;

  &::before {
    background-color: currentColor;
    border-radius: var(--corner-radius);
    bottom: 0;
    content: '';
    left: 0;
    opacity: 0;
    position: absolute;
    right: 0;
    top: 0;
    transition: opacity var(--duration-short);
  }

  &:hover::before,
  &:focus::before {
    opacity: 0.08;
  }
`;

export const IconButton = styled(Button).attrs({
  'aria-label': ({ title }) => title,
})`
  &::before {
    border-radius: 50%;
  }
`;

export const RaisedButton = styled.button.attrs({
  type: 'button',
})`
  ${baseStyles};
  background-color: ${({ red }) =>
    red ? 'var(--color-red)' : 'var(--color-green)'};
  box-shadow: var(--elevation-low);
  color: var(--color-white);
  transition: opacity var(--duration-short);

  &:hover,
  &:focus {
    opacity: 0.64;
  }
`;
