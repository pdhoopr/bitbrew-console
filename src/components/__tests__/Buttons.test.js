import { diffComponents } from 'bitbrew-test-helpers';
import { shallow } from 'enzyme';
import React from 'react';
import styled from 'styled-components';
import { baseStyles, RaisedButton, TextButton } from '../Buttons';

const BaseButton = styled.button`
  ${baseStyles};
`;

test('renders DOM shared by all buttons', () => {
  const dom = shallow(<BaseButton />);
  expect(dom).toMatchSnapshot();
});

test('renders DOM specific to a raised button', () => {
  const base = <BaseButton />;
  const comp = <RaisedButton />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});

test('renders DOM specific to a text button', () => {
  const base = <BaseButton />;
  const comp = <TextButton />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});
