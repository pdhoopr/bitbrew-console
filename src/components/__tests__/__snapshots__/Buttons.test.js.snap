// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM shared by all buttons 1`] = `
.c0 {
  background: none;
  border: none;
  border-radius: var(--corner-radius);
  cursor: pointer;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  font-weight: var(--weight-bold);
  line-height: var(--size-20);
  padding: var(--size-8) var(--size-16);
  text-transform: uppercase;
  -webkit-transition: opacity var(--duration-short);
  transition: opacity var(--duration-short);
}

.c0:hover {
  opacity: 0.8;
}

<button
  className="c0"
/>
`;

exports[`renders DOM specific to a raised button 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -12,6 +12,9 @@
    text-transform: uppercase;
    -webkit-transition: opacity var(--duration-short);
    transition: opacity var(--duration-short);
+   color: var(--color-white);
+   background-color: var(--color-green);
+   box-shadow: var(--elevation-low);
  }
  
  .c0:hover {
`;

exports[`renders DOM specific to a text button 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -12,6 +12,7 @@
    text-transform: uppercase;
    -webkit-transition: opacity var(--duration-short);
    transition: opacity var(--duration-short);
+   color: inherit;
  }
  
  .c0:hover {
`;
